Mathlib/NumberTheory/SiegelsLemma.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2024 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero, Laura Capuano, Amos Turchet
+-/
+import Mathlib.Analysis.Matrix
+import Mathlib.Data.Matrix.Basic
+import Mathlib.Data.Pi.Interval
+
+/-!
+# Siegel's Lemma, first version for ℤ
+
+In this file we introduce and prove Siegel's Lemma in its most basic version. This is a fundamental
+tool in diophantine approximation and transcendency and says that there exists a "small" integral
+non-zero solution of a non-trivial overdetermined system of linear equations with integer
+coefficients.
+
+## Main results
+
+- `siegels_lemma`: Given a non-singular `m × n` matrix `A` with `m < n` the linear system it
+determines has a non-zero integer solution `t` with `‖t‖ ≤ ((n * ‖A‖) ^ ((m : ℝ) / (n - m)))`
+
+## Notation
+
+ - `‖_‖ ` : Matrix.seminormedAddCommGroup is the sup norm, the maximum of the absolute values of
+ the entries of the matrix
+
+## References
+
+See [M. Hindry and J. Silverman, Diophantine Geometry: an Introduction] .
+-/
+
+/- We set ‖⬝‖ to be Matrix.seminormedAddCommGroup  -/
+attribute [local instance] Matrix.seminormedAddCommGroup
+
+open  Finset
+
+/-- sup commutes with casting from Nat to NNReal -/
+lemma cast_sup_eq_sup_cast_Nat_NNReal {S : Type*} (f : S → ℕ) (s : Finset S) :
+    (sup s f) = sup s fun b ↦ (f b : NNReal) :=
+  comp_sup_eq_sup_comp_of_is_total Nat.cast Nat.mono_cast
+  (by simp only [bot_eq_zero', CharP.cast_eq_zero])
+
+namespace Int.Matrix
+
+/-- The definition of ‖⬝‖ for integral matrixes -/
+lemma sup_sup_norm_def (m n : ℕ) (A : Matrix (Fin m) (Fin n) ℤ)  :
```

Comment:
Do you need the matrix to be indexed by `Fin n` and `Fin m` or any pair of `Fintype` would do? It would be closer to the whole library design.

Reply:
I changed it to Fintype as you suggest